test/e2e/velero_utils.go

```diff
@@ -41,20 +41,49 @@ import (
 	veleroexec "github.com/vmware-tanzu/velero/pkg/util/exec"
 )
 
-func getProviderPlugins(providerName string) []string {
-	// TODO: make plugin images configurable
-	switch providerName {
-	case "aws":
-		return []string{"velero/velero-plugin-for-aws:v1.2.1"}
-	case "azure":
-		return []string{"velero/velero-plugin-for-microsoft-azure:v1.2.0"}
-	case "vsphere":
-		return []string{"velero/velero-plugin-for-aws:v1.2.1", "vsphereveleroplugin/velero-plugin-for-vsphere:v1.1.1"}
-	case "gcp":
-		return []string{"velero/velero-plugin-for-gcp:v1.2.1"}
-	default:
-		panic(fmt.Errorf("unknown provider name: %s", providerName))
+var pluginsMatric = map[string]map[string][]string{
+	"v1.4": {
+		"aws":     {"velero/velero-plugin-for-aws:v1.1.0"},
+		"azure":   {"velero/velero-plugin-for-microsoft-azure:v1.1.2"},
+		"vsphere": {"velero/velero-plugin-for-aws:v1.1.0"},
+		"gcp":     {"velero/velero-plugin-for-gcp:v1.1.0"},
+	},
+	"v1.5": {
+		"aws":     {"velero/velero-plugin-for-aws:v1.1.0"},
+		"azure":   {"velero/velero-plugin-for-microsoft-azure:v1.1.2"},
+		"vsphere": {"velero/velero-plugin-for-aws:v1.1.0", "vsphereveleroplugin/velero-plugin-for-vsphere:v1.1.1"},
+		"gcp":     {"velero/velero-plugin-for-gcp:v1.1.0"},
+	},
+	"v1.6": {
+		"aws":     {"velero/velero-plugin-for-aws:v1.2.1"},
+		"azure":   {"velero/velero-plugin-for-microsoft-azure:v1.2.1"},
+		"vsphere": {"velero/velero-plugin-for-aws:v1.2.1", "vsphereveleroplugin/velero-plugin-for-vsphere:v1.1.1"},
+		"gcp":     {"velero/velero-plugin-for-gcp:v1.2.1"},
+	},
+	"v1.7": {
+		"aws":     {"velero/velero-plugin-for-aws:v1.3.0-rc.1"},
+		"azure":   {"velero/velero-plugin-for-microsoft-azure:v1.3.0-rc.1"},
+		"vsphere": {"velero/velero-plugin-for-aws:v1.3.0-rc.1", "vsphereveleroplugin/velero-plugin-for-vsphere:v1.1.1"},
+		"gcp":     {"velero/velero-plugin-for-gcp:v1.3.0-rc.1"},
+	},
+}
+
+func getProviderPlugins(version, providerName string) []string {
+
+	arr := strings.Split(version, ".")
+	if len(arr) < 3 {
+		panic(fmt.Errorf("invalid velero version: %s", version))
```

Comment:
Rather than using panic here and below, we should either return an error, or default to using the plugins for the latest version. It's possible that a local build running this test does not use a semver tag.

Reply:
You're right, I've modified the logic to get plugins by version and provider, if the user input or value from command of "velero version"  is a semver tag and can be located in the matrix, then the right plugins will be returned, otherwise, return the default latest plugins. Thanks!